stdlib/asyncio/events.pyi

```diff
@@ -83,8 +88,8 @@ class TimerHandle(Handle):
     def __init__(
         self,
         when: float,
-        callback: Callable[..., object],
-        args: Sequence[Any],
+        callback: Callable[[Unpack[_Ts]], object],
+        args: tuple[Unpack[_Ts]],
```

Comment:
I'd prefer if we kept `Sequence` here, following the philosophy that false negatives are preferable to false positives. (As lists or other sequences are fine as `args` arguments.)

Reply:
In that case, we can't get type safety though, because only tuples can be heterogeneous in the type system.

Reply:
Unfortunately, I can find a few examples on grep.app of people passing non-tuple sequences to the `args` parameter. People doing things like that will be broken by this change, it's true:
- https://github.com/pulumi/pulumi/blob/273951bb17ef45efeeebfe25be174c6df10c1e08/sdk/python/lib/pulumi/runtime/sync_await.py#L80
- https://github.com/ask/mode/blob/a104009f0c96790b9f6140179b4968da07a38c81/mode/utils/loops.py#L61-L65

I don't see a particularly strong use case for passing a `list` or a `deque` or whatever to `args`, though -- I think it would be fairly trivial, if you _have_ got a non-tuple `Sequence`, to cast it to a `tuple` before passing it to the constructor of `Handle` or `TimerHandle`. And as Jelle says, if we want to use `Sequence` for `args` here, we essentially have to just leave the constructors for `Handle` and `TimerHandle` as they are; you can only use `Unpack` with a `tuple`. If I make this change:

```diff
--- a/stdlib/asyncio/events.pyi
+++ b/stdlib/asyncio/events.pyi
@@ -74,7 +74,7 @@ class Handle:
     def __init__(
         self,
         callback: Callable[[Unpack[_Ts]], object],
-        args: tuple[Unpack[_Ts]],
+        args: Sequence[Unpack[_Ts]],
         loop: AbstractEventLoop,
         context: Context | None = None,
     ) -> None: ...
@@ -89,7 +89,7 @@ class TimerHandle(Handle):
         self,
         when: float,
         callback: Callable[[Unpack[_Ts]], object],
-        args: tuple[Unpack[_Ts]],
+        args: Sequence[Unpack[_Ts]],
         loop: AbstractEventLoop,
         context: Context | None = None,
     ) -> None: ...
```

Then mypy correctly complains:

```
stdlib\asyncio\events.pyi:77: error: Unpack is only valid in a variadic position  [valid-type]
stdlib\asyncio\events.pyi:92: error: Unpack is only valid in a variadic position  [valid-type]
```

Reply:
> In that case, we can't get type safety though, because only tuples can be heterogeneous in the type system.

Yes, that's unfortunate, but requiring `tuple` here is just incorrect. We've always maintained that we prefer false negatives over false positives, unless there's a strong case for doing so. In this case, I consider passing a `list` as completely reasonable.

Reply:
I feel like the benefits in terms of type safety (which seem to me to be large here) outweigh the risk of false-positive errors in this case. However, since this is the only thing you object to in this PR, I'll remove it for now, so that it can be considered separately in its own PR :-)